### ISSUE TITLE
Some issues with RTP plotting

### ISSUE THREAD

**Issue by asreimer** [SuperDARN/pydarn#24]
I'm using the `develop` branch, specifically commit 9b5caa22fda2806d9dbe40146c2403de86255253 and I noticed several small issues. I might be able to help fix them, but figured they should be reported:

## Summary of issues:
- [x] default velocity colormap is reversed
- [ ] when you filter out frequency for for some modes, you get stripes in the RTI
- [x] some documentation for RTP is wrong
- [ ] auto-scale for z goes wacky

## Details
1) The default velocity colormap doesn't follow the redshift/blueshift convention. For example, I plotted up some Saskatoon radar data with this code:

    import pydarn
    from matplotlib import pyplot
    fitted_file = "20120614.0401.1200.sas.lmfit2"
    darn_read = pydarn.SDarnRead(fitted_file)
    data = darn_read.read_fitacf()
    filts = {'max_scalar_filter':{'tfreq': 12500}}
    fig = pyplot.figure(figsize=(10,8))
    ax1 = fig.add_subplot(111)
    pydarn.RTP.plot_range_time(data, parameter='v',
                               beam_num=12, ax=ax1,
                               filter_settings=filts,
                               zmin=-1000, zmax=1000)
    pyplot.show()

You can download the lmfit2 file here: https://github.com/asreimer/lmfit2/blob/master/example/20120614.0401.1200.sas.lmfit2.gz

The plot I get is this: 
![Figure_1](https://user-images.githubusercontent.com/4604819/70732227-e33bd500-1cbc-11ea-9b39-c73f140c4785.png)

But what I should have gotten is a plot where the colormap is reversed. See: https://github.com/asreimer/lmfit2/blob/master/example/no_filter.png or this RTI that was generated using the VT superdarn website: 
![rti_sas](https://user-images.githubusercontent.com/4604819/70732323-14b4a080-1cbd-11ea-9bb8-d66796685949.jpg)

Notice that red and blue are backwards.

2) If you look at the plot I generated using the code in 1, you'll see it has many white stripes. This is undesirable behaviour and will likely happen on all of the "twofsound" mode data.

3) The docstring for RTP says that one should use the `plot_filter` keywork, which is incorrect. One needs to use `filter_settings` instead.

4) The autoscaling for the parameter being plotted can go wacky. Running this code:

    filts = {'min_scalar_filter':{'tfreq': 12500}}
    fig = pyplot.figure(figsize=(10,8))
    ax1 = fig.add_subplot(111)
    pydarn.RTP.plot_range_time(data, parameter='p_l',
                               beam_num=12, ax=ax1,
                               filter_settings=filts)
    
    pyplot.show()

Gives this:
![wat](https://user-images.githubusercontent.com/4604819/70734416-0f595500-1cc1-11ea-98c6-cadfaf706e51.png)

but using the `zmin` and `zmax` fixes the problem:

    
    filts = {'min_scalar_filter':{'tfreq': 12500}}
    fig = pyplot.figure(figsize=(10,8))
    ax1 = fig.add_subplot(111)
    pydarn.RTP.plot_range_time(data, parameter='p_l',
                               beam_num=12, ax=ax1,
                               filter_settings=filts,
                               zmin=1,zmax=30)
    
    pyplot.show()
    
![better](https://user-images.githubusercontent.com/4604819/70734460-25ffac00-1cc1-11ea-8155-d81ffee7ef3b.png)

**Comment by asreimer** [comment]
Found another issue, so I updated the original comment.

**Comment by mts299** [comment]
Thank you for your suggestions. 
1. Range-time doesn't do parameter detection, so if you want a specific colormap for velocity then you can set it with `cmap` The current one was based on how it is not specific for a specific parameter and is color-blind friendly. 

2. The stripping might be caused due to the plot plotting data based on the actual time as per-requested by people. I can look further to determine if there is way around the stripping. 

3. Documentation is being handled in another branch as things keep changing and it is hard to keep remembering to keep up with it. This is already resolved thanks to @billetd 

4. I can look into this.

**Comment by egthomas** [comment]
I agree with @asreimer that the redshift/blueshift convention should be enforced for the default velocity colormap.  That is the standard convention for presenting line-of-sight velocity data measured by SuperDARN (and many other space science instruments), and will likely be a continued source of confusion for science users (particularly those transitioning from davitpy).

**Comment by mts299** [comment]
@egthomas please choose your words wisely. pyDARN will not "enforce" a color map on a user, however, we can change the default colormap. I would like to remind everyone we still do not have standards set for SuperDARN, however, I do know the red-blue shift is a common convention for velocity plotting for most instruments.  Another assumption is DaVitpy != pyDARN, pyDARN has yet to take code from DaVitpy and some of the plotting defaults have changed already.  

I will change the colormap default to the velocity one seen in summary plots as @billetd pointed out majority of scientists will be plotting this parameter. However, this will be the **default for all parameters,** as this keeps with future-proofing. `plot_range_time` is not a function based on specific parameters like DaVitpy and will not do parameter detecting as this will require more maintenance in the future if things change. I will as the main developer and not having a working group to follow rules keep my principles of "Best practices". If people desire this enough then make a branch and create a PR.  

As a reminder about communication, please ask a question about defaults and behaviors.

**Comment by asreimer** [comment]
1. I think most of the issues I raised can be fixed with better
documentation and some clear usage examples. The plot_range_time function
appears to be a general purpose work horse function that simply plots
whatever parameter you specify if it is available in the dmap file for
plotting.

As @egthomas mentioned, this is a very different usage case from what
people who used davitpy are expecting (myself included). Indeed this is
usage case is not entirely self evident just by reading the existing
documentation.

2. I don't think a change to the default colormap is a good idea. As long
at the usage case for plot_range_time is clear, The issue I encountered
won't happen. The choice of colormap probably doesn't matter either way
honestly.


3. I don't see anything wrong with @egthomas comment. I don't think it
warrants the response it received. There are going to be a bunch of people
who are expecting things to work in certain ways, not just from using
davitpy, but that, go (the SuperDARN package that was cool before
Google's), and davit. That is perspective that is valuable and pydarn will
have to adopt and/or parry those expectations. Again with documentation 😉

No rush or anything. I totally understand this takes time.
